exec/bigmachine.go

```diff
@@ -1091,7 +1106,13 @@ func (w *worker) CommitCombiner(ctx context.Context, key TaskName, _ *struct{})
 		case combinerCommitted:
 			return nil
 		case combinerError:
-			return errors.E("error while writing combiner")
+			err := w.combinerErrors[key]
+			if strings.HasPrefix(err.Error(), "gob:") {
+				// No point in retrying any error to do with gob encoding/decoding.
+				// Mark the error as invalid to prevent it from being revised.
+				return errors.E(errors.Fatal, errors.Invalid, fmt.Errorf("fatal error while writing combiner: %w", err))
```

Comment:
`%w` does not compile in Go 1.12, which I'm personally fine with -- we'd just need to turn off the 1.12 CI.

Reply:
In this case it's better to use `errors.E`s mechanism for the same, since it means we can properly deconstruct the errors if needed.

`errors.E(errors.Fatal, "blah", err)`